### PR TITLE
[feature 23] Move the priority card when clicking on it

### PR DESCRIPTION
**Pull request by Vio1625**

### CODE REVIEW

File: src/client/GameWindow.cpp [hunk 1]
```diff
@@ -31,17 +30,22 @@
...
      * \brief Constructor
      *
      * Constructor of GameWindow class
      */
     GameWindow::GameWindow()
     {
+
+        clientGameWindow = std::make_shared<sf::RenderWindow>();
+        clientGameWindow->create(sf::VideoMode(WINDOW_LENGTH, WINDOW_WIDTH), "Civilization VII", sf::Style::Close);
+        clientGameWindow->setPosition(sf::Vector2i(0, 0));
```
Comment (lucas-mrq) [inline src/client/GameWindow.cpp:47]: Already done in the Engine, you should delete line 45 -> 47

File: src/client/GameWindow.cpp [hunk 2]
```diff
@@ -244,9 +253,75 @@ namespace client
...
+    {
+
+        const Json::Value &dataNumber = openJsonFile("/img/hud/data-number.json");
+
+        float priorityScale = dataNumber["priority-card-proportion"].asFloat() / (float(card->texture->getWidth()) / float(WINDOW_LENGTH));
+
+        int yPos = 0;
+        int xPos = 0;
+        int xTitlePos = 0;
+        int xBodyPosition = 0;
```
Comment (lucas-mrq) [inline src/client/GameWindow.cpp:296]: You should not initialized it

File: src/client/GameWindow.cpp [hunk 3]
```diff
@@ -283,73 +370,53 @@ namespace client
...
-        const Json::Value &data = obj["data"];
-
-        return data;
     }
 
     /*!
      * \brief Display text on the cards
      */
-    void GameWindow::displayText(std::vector<CardStruct> *cards, std::string title, std::string body, sf::Font *font)
+    void GameWindow::displayText(std::vector<CardStruct> *cards, std::string title, std::string body, sf::Font *titleFont, sf::Font *bodyFont, float titleTextSizeProportion, float bodyTextSizeProportion)
```
Comment (lucas-mrq) [inline src/client/GameWindow.cpp:383]: Change the name of this function to something more explicit

File: src/client/GameWindow.cpp [hunk 4]
```diff
@@ -283,73 +370,53 @@ namespace client
...
         cards->back().title->setPosition(xTitlePosition, yTitlePosition);
 
         // display the body on the card
-        cards->back().body = (std::unique_ptr<sf::Text>)new sf::Text(body, *font, 30);
+        cards->back().body = (std::unique_ptr<sf::Text>)new sf::Text(body, *bodyFont, bodyTextSize);
 
         // to have the text on several lines without exceeding the card
         int countEndLine = 1;
-        while (cards->back().body->getLocalBounds().width > cards->back().texture->getWidth() - 10)
+        while (cards->back().body->getLocalBounds().width > cards->back().texture->getWidth() - 18) // 18 to not touch the black border
```
Comment (lucas-mrq) [inline src/client/GameWindow.cpp:403]: #define CARD_BORDER 18

File: src/client/GameWindow.cpp [hunk 5]
```diff
@@ -447,6 +514,26 @@ namespace client
...
+    /*!
+     * \brief Get position of number of boxes and boxes on priority cards
+     */
+
+    sf::Vector2i GameWindow::getBoxesElementsPosition(float boxXProportion, float boxYProportion, CardStruct *priorityCard)
+    {
+        int xBoxPos = 0;
+        int yBoxPos = 0;
+        int xBoxOffset = 0;
+        int yBoxOffset = 0;
```
Comment (lucas-mrq) [inline src/client/GameWindow.cpp:526]: You should not initialized it

**Review by lucas-mrq — CHANGES_REQUESTED** [review]

File: src/client/GameWindow.cpp [hunk 6]
```diff
@@ -528,10 +627,10 @@ namespace client
             std::string text = "Player ";
             text += std::to_string(i + 1);
             int offset = dataNumber["offset-between-up-player"].asInt();
-            int upPosition = (WINDOW_LENGTH + (float(2/3) - dataNumber["nb-player"].asInt()) * offset) / 2;
+            int upPosition = (WINDOW_LENGTH + (float(2 / 3) - dataNumber["nb-player"].asInt()) * offset) / 2;
 
-            whoIsPlayingButtons.emplace_back(sf::Vector2f(offset* float(float(2)/float(3)), offset / 2), sf::Vector2f(upPosition + offset * i, 0), PLAYER_COLOR[i], isPlaying);
-            whoIsPlayingButtons.back().setText(dataNumber["up-player-text-size"].asInt(), sf::Vector2f(0, 0), text, &priorityFont);
+            whoIsPlayingButtons.emplace_back(sf::Vector2f(offset * float(float(2) / float(3)), offset / 2), sf::Vector2f(upPosition + offset * i, 0), PLAYER_COLOR[i], isPlaying);
```
Comment (NiskuT) [inline src/client/GameWindow.cpp:632]: Go with several lines

File: src/client/GameWindow.cpp [hunk 7]
```diff
@@ -476,44 +522,97 @@ namespace client
         }
 
         // load the priorityCard
-        if (!priorityFont.loadFromFile(RESOURCES_PATH "/img/hud/font.otf"))
+        boxTexture = (std::unique_ptr<TextureDisplayer>)new TextureDisplayer(RESOURCES_PATH "/img/hud/box.png");
```
Comment (NiskuT) [inline src/client/GameWindow.cpp:525]: std::make_unique without `new`

File: src/client/GraphicCard.cpp [hunk 8]
```diff
@@ -0,0 +1,76 @@
+#include <client.hpp>
+#include <iostream>
+
+#define ACTION_INDEX 11
+#define PRIORITY_INDEX 7
+
+namespace client
+{
+
```
Comment (NiskuT) [inline src/client/GraphicCard.cpp:9]: Write a comment for each constructor to explain that one is used for priorityCards and the other for the other type

**Review by NiskuT — CHANGES_REQUESTED** [review]

File: src/client/GraphicCard.cpp (hunk 8 above)
Comment (Vio1625) [reply src/client/GraphicCard.cpp:9]: It's done in improvement-19

**Review by NiskuT — APPROVED** [review]

**Review by lucas-mrq — APPROVED** [review]

**Pull request merged**